go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -385,3 +405,18 @@ func (oc *BaseSecondaryLayer2NetworkController) syncNodes(nodes []interface{}) e
 
 	return nil
 }
+
+func (oc *BaseSecondaryLayer2NetworkController) syncIPAMClaims(ipamClaims []interface{}) error {
+	switchType := types.OVNLayer2Switch
+	if oc.NetInfo.TopologyType() == types.LocalnetTopology {
+		switchType = types.OVNLocalnetSwitch
+	}
+	if oc.allocatesPodAnnotation() {
```

Comment:
nit: this wouldn't be called on networks that don't support persistent IPs. We don't need this check here. We dont have it on the interconnect side for example

Reply:
Done.